hazelcast/src/main/java/com/hazelcast/cluster/impl/JoinMessage.java

```diff
@@ -34,25 +36,25 @@
     protected String uuid;
     protected boolean liteMember;
     protected ConfigCheck configCheck;
-    protected Collection<Address> memberAddresses;
+    protected Collection<Member> members;
```

Comment:
I see. I missed the place you filtered the members. Or you can filter data-members on sending side and only send addresses of data-members.